File: backend/app/schemas/memory.py
from datetime import datetime

from pydantic import BaseModel, Field


class MemoryChunkRead(BaseModel):
    id: int
    project_id: int
    source_type: str
    source_id: int
    source_title: str
    content: str
    chunk_index: int
    created_at: datetime

    class Config:
        from_attributes = True


class MemoryReindexResponse(BaseModel):
    project_id: int
    chunks_created: int
    documents_indexed: int
    notes_indexed: int
    bugs_indexed: int
    decisions_indexed: int


class MemoryStatsResponse(BaseModel):
    project_id: int
    total_chunks: int
    document_chunks: int
    daily_note_chunks: int
    bug_chunks: int
    decision_chunks: int


class MemorySearchRequest(BaseModel):
    query: str = Field(..., min_length=1)
    top_k: int = Field(default=5, ge=1, le=20)
    source_type: str | None = None


class MemorySearchResult(BaseModel):
    chunk_id: int
    project_id: int
    source_type: str
    source_id: int
    source_title: str
    content: str
    chunk_index: int
    distance: float


class MemorySearchResponse(BaseModel):
    query: str
    top_k: int
    source_type: str | None = None
    results: list[MemorySearchResult]

File: frontend/src/types/memory.ts
export interface MemoryChunk {
  id: number;
  project_id: number;
  source_type: string;
  source_id: number;
  source_title: string;
  content: string;
  chunk_index: number;
  created_at: string;
}

export interface MemoryReindexResponse {
  project_id: number;
  chunks_created: number;
  documents_indexed: number;
  notes_indexed: number;
  bugs_indexed: number;
  decisions_indexed: number;
}

export interface MemoryStats {
  project_id: number;
  total_chunks: number;
  document_chunks: number;
  daily_note_chunks: number;
  bug_chunks: number;
  decision_chunks: number;
}

export interface MemorySearchResult {
  chunk_id: number;
  project_id: number;
  source_type: string;
  source_id: number;
  source_title: string;
  content: string;
  chunk_index: number;
  distance: number;
}

export interface MemorySearchResponse {
  query: string;
  top_k: number;
  source_type: string | null;
  results: MemorySearchResult[];
}
